src/zenml/orchestrators/base_orchestrator.py

```diff
@@ -169,29 +171,41 @@ def run(
         self,
         deployment: "PipelineDeploymentResponse",
         stack: "Stack",
+        placeholder_run: Optional["PipelineRunResponse"] = None,
     ) -> Any:
         """Runs a pipeline on a stack.
 
         Args:
             deployment: The pipeline deployment.
             stack: The stack on which to run the pipeline.
-
-        Returns:
-            Orchestrator-specific return value.
+            placeholder_run: An optional placeholder run for the deployment.
+                This will be deleted in case the pipeline deployment failed.
         """
         self._prepare_run(deployment=deployment)
 
         environment = get_config_environment_vars(deployment=deployment)
 
         try:
-            result = self.prepare_or_run_pipeline(
-                deployment=deployment, stack=stack, environment=environment
-            )
+            if result := self.prepare_or_run_pipeline(
```

Comment:
The name could be clearer IMO, maybe something like `metadata` or `metadata_iterator`?

Reply:
One other thing that we should check here in case you didn't yet: Other orchestrators should not return anything from this method anymore, otherwise it would fail in the iterating here.

Reply:
Yes, I have checked this before and all orchestrators seem to be fine. Applied the renaming suggestion 👍